crates/adapters/src/controller/mod.rs

```diff
@@ -691,75 +690,68 @@ impl CircuitThread {
     /// to send when the records have been processed, and the corresponding
     /// steps log entries.
     fn flush_input_to_circuit(&mut self) -> Result<FlushedInput, ()> {
+        // Collect the ids of the endpoints that we'll flush to the circuit.
```

Comment:
I am not familiar with this code in-depth. Just to make sure: is there any way a connector that is added and immediately removed after sending some data within the same step is somehow missed by this logic?